File: TradeZoneBackend/src/main/java/TradeZone/data/model/entity/Region.java
package TradeZone.data.model.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import java.util.List;

@Entity(name = "regions")
@Getter
@Setter
@NoArgsConstructor
public class Region extends BaseEntity {

    @Column
    private String name;

    @OneToMany(mappedBy = "region")
    private List<Town> towns;
}

File: TradeZoneFrontEnd/src/app/core/models/region.ts
import { TownInRegion } from "./town-in-region";

export class Region {
    name: string;
    totalAds: number;
    totalUsers: number;
    towns: TownInRegion[];
    constructor(name: string, totalAds: number, totalUsers: number) {
        this.name = name;
        this.totalAds = totalAds;
        this.totalUsers = totalUsers;
    }
}
